wrapper/src/main/java/software/amazon/jdbc/plugin/DefaultConnectionPlugin.java

```diff
@@ -204,14 +205,17 @@ private Connection connectInternal(
       }
     }
 
-    this.connProviderManager.initConnection(conn, driverProtocol, hostSpec, props);
+    this.pluginService.setIsPooledConnection(connectionInfo.isPooled());
+    this.connProviderManager.initConnection(connectionInfo.getConnection(), driverProtocol, hostSpec, props);
 
-    this.pluginService.setAvailability(hostSpec.asAliases(), HostAvailability.AVAILABLE);
+    if (connectionInfo.getConnection() != null) {
```

Comment:
I think we can mark this method with a NonNull return value, and then we don't have to make this check

Reply:
Unfortunately connection could be null.